grails-app/services/org/pih/warehouse/core/DocumentTemplateService.groovy

```diff
@@ -192,4 +195,73 @@ class DocumentTemplateService {
         context.put("orderAdjustments", orderAdjustments);
         return context
     }
+
+    // TODO: make it generic for requistions and orders
+    def renderRequisitionDocumentTemplate(Document documentTemplate, Requisition requisitionInstance, ConverterTypeTo targetDocumentType, OutputStream outputStream) {
+        try {
+            InputStream inputStream = new ByteArrayInputStream(documentTemplate.fileContents)
+            IXDocReport report = XDocReportRegistry.getRegistry().loadReport(inputStream, TemplateEngineKind.Freemarker)
+
+            IContext context = createRequisitionContext(report, requisitionInstance)
+
+            ConverterTypeVia sourceDocumentType = (report.kind == "DOCX") ? ConverterTypeVia.DOCX4J :
+                (report.kind == "ODT") ? ConverterTypeVia.ODFDOM : ConverterTypeVia.XWPF
+
+            // Convert to the target type
+            if (targetDocumentType && sourceDocumentType) {
+                Options options = Options.getTo(targetDocumentType).via(sourceDocumentType)
+                report.convert(context, options, outputStream)
+            }
+            // Process document in place
+            else {
+                report.process(context, outputStream)
+            }
+            log.info "Report dump: " + report.dump()
+        } finally {
+            PdfUtil.restoreBaseFonts()
+        }
+    }
+
+    private createRequisitionContext(IXDocReport report, Requisition requisitionInstance) {
+        def requisitionItems = requisitionInstance?.requisitionItems?.collect { RequisitionItem requisitionItem ->
```

Comment:
@jmiranda I had to go with this version either way, because a couple of properties used are transients, and this was easier. I tested it on 150-200 items, and the performance was very good.